### PR TITLE
feat(coinjoin): process dsq

### PR DESCRIPTION
**Pull request by Syn-McJ**
## Issue being fixed or feature implemented
DSQ message has to be processed
DashSync: https://github.com/dashpay/dashsync-iOS/pull/547

## What was done?
- Processing logic
- Bug fixes

## How Has This Been Tested?
N/A

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added or updated relevant unit/integration/functional/e2e tests
- [ ] I have made corresponding changes to the documentation

**For repository code-owners and collaborators only**
- [ ] I have assigned this pull request to a milestone

### CODE REVIEW

File: dash-spv-coinjoin/src/coinjoin_client_queue_manager.rs [hunk 1]
```diff
@@ -48,23 +49,23 @@ impl CoinJoinClientQueueManager {
     }
 
     pub fn check_queue(&mut self) {
-        let current_time = Instant::now().elapsed().as_secs();
-        self.coinjoin_queue.retain(|q| !q.is_time_out_of_bounds(current_time as i64));
+        let current_time = SystemTime::now().duration_since(UNIX_EPOCH).unwrap().as_secs();
+        self.coinjoin_queue.retain(|q| !q.is_time_out_of_bounds(current_time));
```
Comment (Syn-McJ) [inline dash-spv-coinjoin/src/coinjoin_client_queue_manager.rs:53]: I looks like `Instant::now()` doesn't report correct time

File: dash-spv-coinjoin/src/messages/coinjoin_queue_message.rs [hunk 2]
```diff
@@ -89,6 +91,7 @@ impl encode::Decodable for CoinJoinQueueMessage {
         let pro_tx_hash = UInt256::consensus_decode(&mut d)?;
         let time = i64::consensus_decode(&mut d)?;
         let ready: bool = bool::consensus_decode(&mut d)?;
+        let _signature_len = VarInt::consensus_decode(&mut d)?;
         let signature: Option<UInt768> = UInt768::consensus_decode(&mut d).ok();
```
Comment (Syn-McJ) [inline dash-spv-coinjoin/src/messages/coinjoin_queue_message.rs:95]: The issue with signature verification was that the message wasn't parsed correctly (there is the length before the signature)

File: dash-spv-apple-bindings/build.sh [hunk 3]
```diff
@@ -46,12 +46,19 @@ rustup target add x86_64-apple-ios
...
 
-cargo lipo --release
-cargo build --target=x86_64-apple-darwin --release
-cargo build --target=aarch64-apple-darwin --release
-cargo build --target=x86_64-apple-ios --release
-cargo build --target=aarch64-apple-ios --release
-cargo build --target=aarch64-apple-ios-sim --release
+cargo lipo
+cargo build --target=x86_64-apple-darwin 
+cargo build --target=aarch64-apple-darwin
```
Comment (pankcuf) [inline dash-spv-apple-bindings/build.sh:51]: I suppose you just want to make debug symbols accessible in XCode?

In this case you can control it [here](https://github.com/dashpay/dash-shared-core/blob/8a8c03caad1a39965cb4f426c2796a72e241992d/Cargo.toml#L18) by uncommenting `#debug = true` and/or playing around opt-level

File: dash-spv-coinjoin/src/coinjoin_client_queue_manager.rs (hunk 1 above)
Comment (pankcuf) [reply dash-spv-coinjoin/src/coinjoin_client_queue_manager.rs:53]: yes, it's for relative time measurement

**Review by pankcuf — APPROVED** [review]

File: dash-spv-apple-bindings/build.sh (hunk 3 above)
Comment (Syn-McJ) [reply dash-spv-apple-bindings/build.sh:51]: Ok, thanks, that'll be more convenient

**Pull request merged**